library/core/src/ptr/const_ptr.rs

```diff
@@ -152,18 +152,27 @@ impl<T: ?Sized> *const T {
 
     /// Gets the "address" portion of the pointer.
     ///
-    /// This is equivalent to `self as usize`, which semantically discards
-    /// *provenance* and *address-space* information. To properly restore that information,
-    /// use [`with_addr`][pointer::with_addr] or [`map_addr`][pointer::map_addr].
+    /// This is similar to `self as usize`, which semantically discards *provenance* and
+    /// *address-space* information. However, unlike `self as usize`, it is not possible to cast the
+    /// returned address back to a dereferencable pointer via `as *const T`. To properly restore the
+    /// lost information and obtain a dereferencable pointer, use [`with_addr`][pointer::with_addr]
+    /// or [`map_addr`][pointer::map_addr].
+    ///
+    /// If using those APIs is not possible because there is no way to preserve a pointer with the
+    /// required provenance, use [`expose_addr`][pointer::expose_addr] and
+    /// [`from_exposed_addr`][pointer::from_exposed_addr] instead. However, note that this makes
+    /// your code less portable and less amenable to tools that check for compliance with the Rust
+    /// memory model.
     ///
     /// On most platforms this will produce a value with the same bytes as the original
     /// pointer, because all the bytes are dedicated to describing the address.
     /// Platforms which need to store additional information in the pointer may
     /// perform a change of representation to produce a value containing only the address
     /// portion of the pointer. What that means is up to the platform to define.
     ///
-    /// This API and its claimed semantics are part of the Strict Provenance experiment,
-    /// see the [module documentation][crate::ptr] for details.
+    /// This API and its claimed semantics are part of the Strict Provenance experiment, and as such
+    /// might change in the future (including possibly making this wholly equivalent to `self as
+    /// usize`). See the [module documentation][crate::ptr] for details.
```

Comment:
I wonder if this should try to make it clear if such a change would be a strict weakening, or if you're genuinely allowing for 'as usize' to become more strict.

Reply:
The intention was for this to be a weakening. I made the text even more explicit.